website/docs/vscode.md

```diff
@@ -6,231 +6,457 @@ sidebar_label: Visual Studio Code
-the restart.
+- Start the server process using a custom, local build of the Sorbet binary.
 
-If you see an error that looks like this:
+When setting this setting, it's recommended to refer to the more complete
+documentation in the Sorbet extension's
+[package.json](https://github.com/sorbet/sorbet/blob/master/vscode_extension/package.json)
+file.
 
-```plaintext
-[Error - 9:36:32 AM] Connection to server got closed. Server will not be restarted.
-Running Sorbet LSP with:
-    bundle exec srb typecheck --lsp
-Could not locate Gemfile or .bundle/ directory
-```
+### `sorbet.userLspConfigs`
+
+This setting is largely the same as `sorbet.lspConfigs`, but has higher
+precedence. It's designed to allow users to keep their own Sorbet configurations
+in VS Code's User settings.
+
+Normally, Workspace Settings
+[take precedence over](https://code.visualstudio.com/docs/getstarted/settings#_settings-precedence)
+User Settings. Having this option gives users the chance to invert that.
 
-...then you probably need to run `bundle install` to install Sorbet.
+Because of this, it's usually not recommended to commit this setting to
+`.vscode/settings.json` in a workspace, because it prevents users from launching
+Sorbet according to their own preferences.
+
+### `sorbet.selectedLspConfigId`
+
+Which LSP configuration to pick first (see
+[`sorbet.lspConfigs`](#sorbet-lspconfigs)).
+
+If unset, picks the first one, giving precedence to `sorbet.userLspConfigs`.
+
+Remember that [settings are sticky](#sticky-configuration), so if the user has
+used [`>Sorbet: Configure`] to change the configuration, that most recently
+selected configuration will take precedence over this setting.
+
+### `sorbet.highlightUntyped`
+
+Whether to highlight untyped, and where. Defaults to `"nowhere"`.
+
+See [Highlighting untyped code](highlight-untyped.md)
+
+### `sorbet.typedFalseCompletionNudges`
+
+Whether to enable `# typed: false` completion nudges.
+
+See [``>Sorbet: Toggle the autocomplete nudge in `typed: false` files``].
+
+### `sorbet.configFilePatterns`
+
+A list of workspace file patterns that contribute to Sorbet's configuration.
+Changes to any of those files should trigger a restart of any actively running
+Sorbet language server.
+
+For example, changes to the `sorbet/config` file or the `Gemfile.lock` could
+mean that the Sorbet process needs to restart (to pick up new options, or a new
+version, respectively). The default `sorbet.configFilePattern` tracks these two
+files are tracked
+
+Projects that have custom scripts that launch Sorbet likely want to set this
+setting.
+
+Careful: setting this setting completely overwrites the default.
+
+### `sorbet.revealOutputOnError`
+
+Shows the extension output window on errors. Defaults to `false`.
+
+---
+
+## Troubleshooting and FAQ
+
+<!-- TODO(jez) Make separate pages for these features, and document them there.
+Most of these are not VS Code-specific. -->
 
 ### Diagnostics / error squiggles
 
```

Comment:
For ` Sorbet is reporting diagnostics (error squiggles) that don't make sense.`, should we also mention running Sorbet from the command line as a way to check if it's general Sorbet bug or a VSCode specific issue?